### PR TITLE
Remove `conda.cli.common.ensure_name_or_prefix`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Previously an explicit check/exception is defined to ensure that `conda create` includes either `--name` or `--prefix`. These two arguments are already defined as a mutex group in `argparse`. Just set this group to required and `argparse` will do the rest for us.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/master/news/TEMPLATE)) for the next release's release notes?
     
- [X] ~Add / update necessary tests?~
- [X] ~Add / update outdated documentation?~

### CODE REVIEW

**Review by beeankha — APPROVED** [review]
👍

File: conda/cli/common.py [hunk 1]
```diff
@@ -67,6 +68,10 @@ def confirm_yn(message="Proceed", default='yes', dry_run=NULL):
 
 
 def ensure_name_or_prefix(args, command):
+    warn(
+        f"{__name__}.ensure_name_or_prefix is pending deprecation in 5.0.0.",
```
Comment (jezdez) [inline conda/cli/common.py:72]:
```suggestion
        "conda.cli.common.ensure_name_or_prefix is pending deprecation in a future release.",
```

File: news/11490-deprecate-conda.cli.common.ensure_name_or_prefix [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+* <news item>
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* `conda.cli.common.ensure_name_or_prefix` is pending deprecation in 5.0.0. (#11490)
```
Comment (jezdez) [inline news/11490-deprecate-conda.cli.common.ensure_name_or_prefix:11]:
```suggestion
* `conda.cli.common.ensure_name_or_prefix` is pending deprecation in a future release. (#11490)
```

**Review by jezdez — APPROVED** [review]
Just two nits about not setting the release policy in stone yet before the CEP is being discussed and voted on.

**Pull request merged**